### PR TITLE
Add facial recognition when the device allows. Add Brazilian Language. fix #60, fix #268

### LINKED ISSUE #60: 2FA displays facial recognition when the device allows
Reference article
https://medium.com/p/707e7db17edc/responses/show

### LINKED ISSUE #268: Add Portuguese Brazilian language option
**Is your feature request related to a problem? Please describe.**
Portuguese Brazilian language option is missing.

**Describe the solution you'd like**
Add Portuguese Brazilian language option to language page

**Describe alternatives you've considered**

**Additional context**
(optional: include links to other issues, PRs, tickets, etc)

### PR DESCRIPTION
**Pull request by AmazingBeerBelly**
This PR #154 is not available because the branch seems to have been deleted.

1. Add Face ID recognition

Kapture 2020-07-09 at 21 32 50

2. Add Brazilian Portuguese Option

截屏2020-07-09 下午9 28 58
截屏2020-07-09 下午9 29 10
截屏2020-07-09 下午9 29 25

### CODE REVIEW

File: src/common/translations/ptBR.json [hunk 1]
```diff
@@ -194,6 +194,20 @@
...
+    "dapp": {
+      "title": "DApp",
+      "search": "Procurem ou introduzem o URL do site",
+      "recent": "Recente",
+      "recommended": "Recomendado",
+      "all": "Todos",
+      "button": {
+        "seeAll": "Veja tudo",
+        "confirm": "Confirmar",
+        "understood": "Entendido",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:206]:
```suggestion
        "understood": "Entendi",
```

File: src/common/translations/ptBR.json [hunk 2]
```diff
@@ -194,6 +194,20 @@
...
       }
     },
+    "dapp": {
+      "title": "DApp",
+      "search": "Procurem ou introduzem o URL do site",
+      "recent": "Recente",
+      "recommended": "Recomendado",
+      "all": "Todos",
+      "button": {
+        "seeAll": "Veja tudo",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:204]:
```suggestion
        "seeAll": "Ver tudo",
```

File: src/common/translations/ptBR.json [hunk 3]
```diff
@@ -194,6 +194,20 @@
         "failed": "Falhou! Seu apelido não foi registrado."
       }
     },
+    "dapp": {
+      "title": "DApp",
+      "search": "Procurem ou introduzem o URL do site",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:199]:
```suggestion
      "search": "Faça uma busca ou insira a URL do site",
```

File: src/common/translations/ptBR.json [hunk 4]
```diff
@@ -219,7 +233,8 @@
         "title": "Autenticação de dois fatores",
         "setPasscode": "Configurar PIN",
         "resetPasscode": "Alterar PIN",
-        "useFingerprint": "Usar impressão digital"
+        "useFingerprint": "Usar impressão digital",
+        "useFaceID": "Usar ID da cara"
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:237]:
```suggestion
        "useFaceID": "Usar Face ID"
```

File: src/common/translations/ptBR.json [hunk 5]
```diff
@@ -403,13 +419,21 @@
...
     "touchSensor": {
-      "title": "Sensor biométrico",
       "usePasscode": "Usar PIN",
-      "touchToVerify": "Toque para verificar a impressão digital",
-      "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+      "fingerprint": {
+        "title": "Sensor biométrico",
+        "touchToVerify": "Toque para verificar a impressão digital",
+        "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+        "notMatch": "A impressão digital não corresponde. Por favor, tente novamente."
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:427]:
```suggestion
        "notMatch": "A impressão digital não confere. Por favor, tente novamente."
```

File: src/common/translations/ptBR.json [hunk 6]
```diff
@@ -403,13 +419,21 @@
...
+      },
+      "faceID": {
+        "title": "ID da cara",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:430]:
```suggestion
        "title": "Face ID",
```

File: src/common/translations/ptBR.json [hunk 7]
```diff
@@ -403,13 +419,21 @@
...
+        "touchToVerify": "Toque para verificar o ID do rosto",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:431]:
```suggestion
        "touchToVerify": "Toque para verificar o Face ID",
```

File: src/common/translations/ptBR.json [hunk 8]
```diff
@@ -431,6 +455,17 @@
...
+    "dappWarning": {
+      "title": "Redireccionar para o Dapp de terceiros \"%{dappName}\"",
+      "body": "%{description}\n\nNota: Você será redirecionado para um tripartite Dapp \"%{dappName}\". Por favor, analise cuidadosamente o Acordo de Utilizador e a Política de Privacidade do Dapp terceiro. \"%{dappName}\" reservará toda a responsabilidade por todas as ações e uso no Dapp.\n\n"
+    },
+    "walletSelection": {
+      "title": "Por favor escolha uma carteira"
+    },
+    "noWalletAvailable": {
+      "title": "Nenhuma carteira disponível",
+      "body": "Ainda não há carteira disponível para esse Dapp.Por favor, navegue para a página Creação de carteira para criar ou importar uma carteira primeiro.\n\n"
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:467]:
```suggestion
      "body": "Ainda não há carteira disponível para esse Dapp. Por favor, primeiro navegue para a página criação de carteira para criar ou importar uma.\n\n"
```

File: src/common/translations/pt.json [hunk 9]
```diff
@@ -407,13 +408,21 @@
...
     "touchSensor": {
-      "title": "Usar senha",
       "usePasscode": "Sensor de toque",
-      "touchToVerify": "Toque para verificar a impressão digital",
-      "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+      "fingerprint": {
+        "title": "Usar senha",
+        "touchToVerify": "Toque para verificar a impressão digital",
+        "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+        "notMatch": "A impressão digital não corresponde. Por favor, tente novamente."
```
Comment (bitcoinuser) [inline src/common/translations/pt.json:416]:
```suggestion
        "notMatch": "A impressão digital não confere. Por favor, tente novamente."
```

File: src/common/translations/ptBR.json [hunk 10]
```diff
@@ -431,6 +455,17 @@
       "title": "Token duplicado",
       "body": "Este token já foi adicionado à carteira. Por favor, tente um token diferente."
     },
+    "dappWarning": {
+      "title": "Redireccionar para o Dapp de terceiros \"%{dappName}\"",
+      "body": "%{description}\n\nNota: Você será redirecionado para um tripartite Dapp \"%{dappName}\". Por favor, analise cuidadosamente o Acordo de Utilizador e a Política de Privacidade do Dapp terceiro. \"%{dappName}\" reservará toda a responsabilidade por todas as ações e uso no Dapp.\n\n"
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:460]:
```suggestion
      "body": "%{description}\n\nNota: Você será redirecionado para um Dapp de terceiros \"%{dappName}\". Por favor, analise cuidadosamente o Termo de Acordo do Usuário e a Política de Privacidade do Dapp. \"%{dappName}\" é responsável por todas as ações e uso no Dapp.\n\n"
```

Comment (bitcoinuser) [reply src/common/translations/ptBR.json:460]: @solangegueiros Essa parte de (É responsávek por todas) faz sentido para você?

File: src/common/translations/ptBR.json [hunk 11]
```diff
@@ -403,13 +419,21 @@
...
+        "title": "Sensor biométrico",
+        "touchToVerify": "Toque para verificar a impressão digital",
+        "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+        "notMatch": "A impressão digital não corresponde. Por favor, tente novamente."
+      },
+      "faceID": {
+        "title": "ID da cara",
+        "touchToVerify": "Toque para verificar o ID do rosto",
+        "nativeNote": "Por favor, apontem para a câmera para o reconhecimento do rosto",
+        "notMatch": "FaceID não marcha.Por favor, tente de novo."
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:433]:
```suggestion
        "notMatch": "Face ID não confere. Por favor, tente de novo."
```

File: src/common/translations/pt.json [hunk 12]
```diff
@@ -407,13 +408,21 @@
...
+      },
+      "faceID": {
+        "title": "ID do rosto",
```
Comment (bitcoinuser) [inline src/common/translations/pt.json:419]:
```suggestion
        "title": "Face ID",
```

File: src/common/translations/ptBR.json [hunk 13]
```diff
@@ -403,13 +419,21 @@
...
+      "fingerprint": {
+        "title": "Sensor biométrico",
+        "touchToVerify": "Toque para verificar a impressão digital",
+        "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+        "notMatch": "A impressão digital não corresponde. Por favor, tente novamente."
+      },
+      "faceID": {
+        "title": "ID da cara",
+        "touchToVerify": "Toque para verificar o ID do rosto",
+        "nativeNote": "Por favor, apontem para a câmera para o reconhecimento do rosto",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:432]:
```suggestion
        "nativeNote": "Por favor, aponte para a câmera para o reconhecimento do rosto",
```

File: src/common/translations/ptBR.json [hunk 14]
```diff
@@ -431,6 +455,17 @@
       "title": "Token duplicado",
       "body": "Este token já foi adicionado à carteira. Por favor, tente um token diferente."
     },
+    "dappWarning": {
+      "title": "Redireccionar para o Dapp de terceiros \"%{dappName}\"",
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:459]:
```suggestion
      "title": "Redirecionando para o Dapp de terceiros \"%{dappName}\"",
```

File: src/common/translations/ptBR.json [hunk 15]
```diff
@@ -403,13 +419,21 @@
...
+        "title": "Sensor biométrico",
+        "touchToVerify": "Toque para verificar a impressão digital",
+        "nativeNote": "Digitalize sua impressão digital no scanner do dispositivo para continuar",
+        "notMatch": "A impressão digital não confere. Por favor, tente novamente."
+      },
+      "faceID": {
+        "title": "Face ID",
+        "touchToVerify": "Toque para verificar o Face ID",
+        "nativeNote": "Por favor, aponte para a câmera para o reconhecimento do rosto",
+        "notMatch": "Face ID não confere. Por favor, tente de novo."
```
Comment (bitcoinuser) [inline src/common/translations/ptBR.json:433]:
```suggestion
        "notMatch": "Face ID não confere. Por favor, tente de novamente."
```

File: src/components/common/modal/touchSensorModal.js [hunk 16]
```diff
@@ -157,12 +159,14 @@ export default class TouchSensorModal extends Component {
   }
 
   renderModal() {
-    const { fingerprintFallback } = this.props;
-    const { errorMessage } = this.state;
+    const { fingerprintPasscodeDisabled, fingerprintFallback } = this.props;
+    const { errorMessage, biometryType } = this.state;
+    const titleText = biometryType === 'Face ID' ? 'modal.touchSensor.faceID.title' : 'modal.touchSensor.fingerprint.title';
```
Comment (chrisli30) [inline src/components/common/modal/touchSensorModal.js:164]: 'Face ID' 这种常量，需要在常量文件里定义，因为可以避免手打出错的问题，另外在常量文件里可以统一查找修改

**Pull request merged**